### DISCUSSION TITLE
Adonis 5 + ReactJs

### DISCUSSION THREAD

**Discussion by brenofreire** [adonisjs/core#1376] [Help] [answered]
Maybe I got it wrong. But is there anyway I can serve and render a react app with Adonis?

We have edge but I don't know how to make it work with edge

**Comment by thetutlage** [comment, marked as answer]
Nope, we do not provide any helpers or methods to render React components. You can use standard react libraries to achieve that. However, I personally recommend everyone to keep the frontend and the backend separate.

Your frontend code should just rely on the HTTP APIs for the data and then you can use AdonisJS to develop that API.

**Reply by adegbengaagoro** [reply]
Hi @thetutlage 

Would the use of Content Negotiation go against your disposition of separating the frontend and the backend?

I ask because I intend developing a full-stack app while using Content Negotiation to still expose a JSON output which a mobile app is to consume.

Thanks

**Reply by iammrsea** [reply]
Isn't your use-case the point of Content Negotiation anyway?
